### ISSUE TITLE
New exchange FTX

### ISSUE THREAD

**Issue by Dante2333** [freqtrade/freqtrade#2443]
Ftx.com
Ftx uses margin and only has a few spots markets. The interesting thing is that we can trade a whole basket of futures. It was added on CCXT already

https://github.com/ccxt/ccxt/pull/5807

How can would it be to support margin? Account have the margin settings to cross meaning that we could just use nominal values and that's it as the exchange will monitor the margin and so on, so if you open 1000 usd position and it loses 1% you'll lose that amount in a net amount from 1000.

**Comment by hroff-1902** [comment]
* it's not in our version of ccxt yet, impossible to make experiments yet, should be there next Monday
* ftx is not the first exchange amongst ccxt exchanges that has futures and margins

**Comment by xmatthias** [comment]
well everyone can install the latest version of ccxt and experiment with it... 

However, as mentioned in #2312, i don't think that adding margins is this easy and nor straightforward and makes a ton of sense now, especially since ccxt does not unify margin trading yet (https://github.com/ccxt/ccxt/issues/4834).

It'll be a hell to maintain since we would need to figure out all custom exchange parameters ourselfs (that's what we have ccxt for).

Also, #2183 is closely related (and work on margin trading will surely NOT start before that is implemented) - since i guess by using margin trading you'd also want to use shorts.

In short, i'm going to close this in favor of #2312 for Margin trading.
If margin trading is going to be supported, then surely for an exchange we already support (binance) first.

for anything else, I don't think that FTX is interresting for freqtrade at the moment, since it only has a handfull of spot-markets, and is not even in the list of the [top 100 exchanges](https://coinmarketcap.com/rankings/exchanges/), so supporting it now would be a waste of developer time.

**Comment by Dante2333** [comment]
@xmatthias can we re-open this? FTX includes multiple spot markets which involve their leveraged tokens which could be traded and backtested using spot markets, I could help debug freqtrade for it. 

https://ftx.com/tokens 
what are leveraged tokens -> https://help.ftx.com/hc/en-us/articles/360025298311-Leveraged-Token-Specs

**Comment by xmatthias** [comment]
i don't see why.
as mentioned above:

> well everyone can install the latest version of ccxt and experiment with it...

If it's spot markets, feel free to experiment with it - in theory it should work without extra  effort.

If it involves margin trading or short or whatever - then we have other issues which track progress - since we'll start with binance (one of the top 10 exchanges by volume) - and definitely not with a new exchange).

If it requires changes to freqtrade, feel free to adapt freqtrade if necessary and make a Pull Request so everyone in the community will benefit.

However, as mentioned above, FTX is a new, small exchange. I do not intend to support it anytime soon (not unless it reaches a certain maturity level for sure).
So: 
* if it works "out of the box" - that's perfect. 
* If it requires changes and you're willing and have the knowledge to contribute these - perfect
* if it requires changes and you want someone else to step in -  i'm sorry but please forget this exchange.

I'm sure everyone finds something interresting in most other new exchanges too (otherwise why create a new exchange in the first place) - if we start ~~wasting~~ investing time to make freqtrade compatible with every new exchange, then we'll not be able to support exchanges which do have the volume to properly trade with freqtrade.

Thanks for your understanding.

**Comment by Dante2333** [comment]
Why do you say that ftx is just some random exchange when:

Ftx was incubated by Alameda Research, who holds 2 spots on bitmex leaderboard and are known market maker / Otc providers in the space. 
Ftx volume accounts for roughly 5% of real volume
FTX growth is unparalleled, it only has 7 months and contains markets that did not exist in the Crypto space before along with perpetual contracts never seen. 

I'll try using spot markets and report back.

**Comment by hroff-1902** [comment]
Well, I'm interested in Freqtrade working with BitMEX, for example. I'm working on it (lazy on background) and I WILL support it when it will be ready. At least that's what I'm going to do.

I'm personally not interested in supporting FTX, for example. So I'm not going to spend my personal resources to resolve possible problems with this exchange and support this then.

Same/similar for @xmatthias as far as I understand.

You can find a person with enough python coding skills which will join the team and support this exchange.

Or you can contact the FTX team, if they are interested in Freqtrade team declaring support for their exchange -- they could provide us a kind of grant for adding the support for their exchange, we could manage it in this case, I guess, why not...

**Comment by xmatthias** [comment]
> Or you can contact the FTX team, if they are interested in Freqtrade team declaring support for their exchange -- they could provide us a kind of grant for adding the support for their exchange, we could manage it in this case, I guess, why not...

i'm not sure about this - we're not commercially supported so far by the supported exchanges - so adding this option for exchanges may not be a "so good" option. this should be very carefully considered.

> I'm personally not interested in supporting FTX, for example. So I'm not going to spend my personal resources to resolve possible problems with this exchange and support this then.

Completely agree. 

> Why do you say that ftx is just some random exchange

Honestly, this (whole thread...) sounds to me like you're somehow affiliated with FTX ... either through some partner program, or directly as an employee. 
Don't get me wrong, that's just a feeling I get from the communications (mainly in this thread) ... 

Either way it won't influence our opinion about supporting another exchange which may or may not have problems (we don't know), when time is already tight to implement functions which would benefit all exchanges.

Feel free to test this exchange (since it's in ccxt, you are able to do that all along), and open issues for problems that pop up - if it's minor problems we will probably be able to fix them rather easily unless it uses trading-methods we don't support yet (margin trading, ...).

**Comment by Dante2333** [comment]
I want to mainly focus on spot trading. I'm just doing the same thing as I did when I found freqtrade. I've suggested to anyone who used a closed source code to migrate to freqtrade because freqtrade was great.

Bitmex is NOT great. I've been trading crypto since 2017, I know about it since 2011 and i've followed it ever since. Bitmex overloads (api issues when overload happens), bitmex altcoin trading is manipulated, from time to time you'll get insane wicks that did not happen on spot markets. Bitmex insurance fund has grown massively as they reap the profits from their overload, slippage and raping their own users through shady stuff. I've traded on bitmex for 2 years and suffered and had to adjust my trading style to do so.

FTX has perpetuals on a lot of altcoins, USD collateral, and OTC platform that can handle huge volumes, a market for spot with leveraged trading products. Leveraged tokens can be used with freqtrade because they are margined spot markets, basically they replicate movements 10x and so on so basically we can trade those without having to use margin, which adds the possibility for cointegration. 

FTX just as freqtrade is great, they have done great things against issues that have arised and listen to the community, try to get Arthur Hayes to listen to any of our suggestions and let me know how that goes.
To list a few events:
Issues happened with orders back in November, everyone affected got refunded, their ccxt support was worked fast, they care about their community and there's nothing that we can not verify. They are registered in Antigua and Barbuda, their headquarters are in Hong Kong and they are trying to get licenses that are just coming to existing so they are regulated.

So its not that im affilited with them, its just hat AS freqtrade is a great product FTX is a great exchange.
